test/main.sh.bash-completion

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# Bash completion for test/main.sh
+#
+# To enable tab completion for the test runner, source this file:
+#   source test/main.sh.bash-completion
+#
+# Or add it to your ~/.bashrc:
+#   echo 'source ~/git/lxd/test/main.sh.bash-completion' >> ~/.bashrc
+
+_lxd_test_completion() {
+    local cur="${COMP_WORDS[COMP_CWORD]}"
+
+    # Special keywords
+    local keywords="all cluster standalone test-shell"
+
+    # Extract all test function names from main.sh
+    # Remove 'test_' prefix and 'run_test test_' wrapper
+    local script_dir
+    script_dir="$(dirname "${BASH_SOURCE[0]}")"
```

Comment:
We already set `MAIN_DIR` in `test/main.sh`.

Reply:
Maybe we use it differently. How else than `make test-shell` do you use those completions?

Reply:
> but others may run ./main.sh from the main shell on their host

Indeed I do

Reply:
Thank you both. I've reworked #16664 to remain compatible with direct invocations of `./main.sh`.